compiler/noirc_frontend/src/hir/comptime/interpreter/foreign.rs

```diff
@@ -31,10 +31,11 @@ fn poseidon2_permutation(
     arguments: Vec<(Value, Location)>,
     location: Location,
 ) -> IResult<Value> {
-    let (input, state_length) = check_two_arguments(arguments, location)?;
+    let ((input, input_location), (state_length, state_length_location)) =
+        check_two_arguments(arguments, location)?;
 
-    let (input, typ) = get_array(interner, input, location)?;
-    let state_length = get_u32(state_length, location)?;
+    let (input, typ) = get_array(interner, input, input_location)?;
+    let state_length = get_u32(state_length, state_length_location)?;
```

Comment:
Or maybe `get_u32`, `get_array`, etc., could take a `(Value, Location)` tuple instead of two separate arguments. I'll try that approach first.

Reply:
Done. I think it looks okay, but let me know and I can change it back to `check_argument_count`. I mainly wanted to avoid having to pop arguments and get them in backwards order... but maybe it's not that confusing.